### PR TITLE
docs(vpc): update VPC documentation in README and API

### PR DESCRIPTION
**Pull request by rix0rrr**
Try to address common gotchas with the VPC component in the
package documentation.

Addresses part of #3704.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/README.md [hunk 1]
```diff
@@ -24,32 +24,103 @@ import ec2 = require('@aws-cdk/aws-ec2');
...
+* **Isolated** - isolated subnets do not route from or to the Internet, and
+  as such do not require NAT gateways. They can only connect to or be
+  connected to from other instances in the same VPC. A default VPC configuration
+  will not include isolated subnets,
+
+A default VPC configuration will create public and private subnets, but not
+isolated subnets. See *Advanced Subnet Configuration* below for information
+on how to change the default subnet configuration.
+
+Classes that use the VPC will "launch instances" (or more accurately, create
```
Comment (nija-at) [inline packages/@aws-cdk/aws-ec2/README.md]: "Classes that use the VPC" -- not clear here. Perhaps you mean 'CDK constructs'?

File: packages/@aws-cdk/aws-ec2/README.md [hunk 2]
```diff
@@ -24,32 +24,103 @@ import ec2 = require('@aws-cdk/aws-ec2');
...
+Elastic Network Interfaces) into one or more of the subnets. They all accept
+a property called `subnetSelection` (sometimes called `vpcSubnets`) to allow
+you to select in what subnet to place the ENIs, usually defaulting to
+*private* subnets if the property is omitted.
+
+If you would like to save on the cost of NAT gateways, you can either switch
```
Comment (nija-at) [inline packages/@aws-cdk/aws-ec2/README.md]: This paragraph jumped around a bit and it took me a few reads to understand what you were saying. Better framing here.

Suggestion: "Using or switching to the isolated subnet type reduces infrastructure cost by avoiding the need for NAT gateways. You could also reduce cost by reducing the number of NAT gateways on your private subnet type by setting the natGateways property to a lower number (the default is ???). Be aware ..."

File: packages/@aws-cdk/aws-ec2/README.md [hunk 3]
```diff
@@ -24,32 +24,103 @@ import ec2 = require('@aws-cdk/aws-ec2');
...
-If you require the ability to configure subnets the `Vpc` can be
-customized with `SubnetConfiguration` array. This is best explained by an
-example:
+By default, a VPC will spread over at most 3 Availability Zones available to
+it. To change the number of Availability Zones that the VPC will spread over,
+specify the `maxAzs` property when defining it.
 
-```ts
-import ec2 = require('@aws-cdk/aws-ec2');
+How many Availability Zones are available depends on the *region* and
```
Comment (nija-at) [inline packages/@aws-cdk/aws-ec2/README.md]: 1. Minor rephrase on 1st line: "The number of availability zones that are available depend on ..."

2. I was aware that AZs are dependent on the region but didn't know they were dependent on the account. Are they?

3. Link to this page would be useful - https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#using-regions-availability-zones-describe

File: packages/@aws-cdk/aws-ec2/README.md [hunk 4]
```diff
@@ -226,27 +200,39 @@ const vpc = new ec2.Vpc(this, 'TheVPC', {
...
 
-#### Importing an existing VPC
+### Importing an existing VPC
+
+If your VPC is created outside your CDK app, you can use `Vpc.fromLookup()`.
+The CDK CLI will search for the specified VPC in the the stack's region and
+account, and import the subnet configuration. Looking up can be done by VPC
+ID, but more flexibly by searching for a specific tag on the VPC.
+
+The import does assume that the VPC will be *symmetric*, i.e. that there are
```
Comment (nija-at) [inline packages/@aws-cdk/aws-ec2/README.md]: What happens if this assumption is not true? Specify that.

**Comment by rix0rrr** [comment]
Fixes #3810.

File: packages/@aws-cdk/aws-ec2/README.md (hunk 3 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-ec2/README.md]: 2. Yes, although this mostly only happens when new AZs are introduced and there is a whitelisting period.

**Pull request merged**